### PR TITLE
JAMES-3754 RFC-5819 IMAP4 Extension for Returning STATUS Information in Extended LIST

### PR DESCRIPTION
**Pull request by chibenwa**
Lunch PR...

### CODE REVIEW

**Comment by chibenwa** [comment]
```
19:14:49,626 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check (check-style) on project protocols-imap: You have 1 Checkstyle violation. -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.2:check (check-style) on project protocols-imap: You have 1 Checkstyle violation.
```

File: protocols/imap/src/main/java/org/apache/james/imap/encode/ListingEncodingUtils.java [hunk 1]
```diff
@@ -19,13 +19,16 @@
...
 package org.apache.james.imap.encode;
 
+import static org.apache.james.imap.api.ImapConstants.NAME_ATTRIBUTE_SUBSCRIBED;
+
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 
 import org.apache.james.imap.api.ImapCommand;
 import org.apache.james.imap.api.ImapConstants;
 import org.apache.james.imap.api.process.MailboxType;
```
Comment (vttranlina) [inline protocols/imap/src/main/java/org/apache/james/imap/encode/ListingEncodingUtils.java:29]: remove unused:
`import org.apache.james.imap.api.process.MailboxType;`

**Comment by vttranlina** [comment]
let me rebase

File: protocols/imap/src/main/java/org/apache/james/imap/decode/parser/ListCommandParser.java [hunk 2]
```diff
@@ -149,6 +151,21 @@ private Pair<ListReturnOption, Optional<StatusDataItems>> readStatus(ImapRequest
         return Pair.of(ListReturnOption.STATUS, Optional.of(StatusCommandParser.statusDataItems(request)));
     }
 
+    private ListReturnOption readSpecialUse(ImapRequestLineReader request) throws DecodingException {
+        // 'S' is already consummed
+        assertChar(request, 'P', 'p');
+        assertChar(request, 'E', 'e');
+        assertChar(request, 'C','c');
```
Comment (quantranhong1999) [inline protocols/imap/src/main/java/org/apache/james/imap/decode/parser/ListCommandParser.java:158]:
```suggestion
        assertChar(request, 'C', 'c');

**Review by quantranhong1999 — COMMENTED** [review]
nothing to add

**Comment by vttranlina** [comment]
The "manually" rebase has been created: https://github.com/apache/james-project/pull/1342

**Pull request closed without merging**